tools/decompile.py

```diff
@@ -52,23 +52,19 @@ def __init__(self, func_root: Path, function_path: Path, args: argparse.Namespac
     def overlay(self) -> str:
         """Attempts to infer the overlay from the asm file path or returns None if one is not found.
         The '_psp' extension will be included if the version is psp"""
-        nonmatchings_dir = (
-            "f_nonmat"
-            if args.version == "saturn"
-            else "psp" if "psp" in args.version else "nonmatchings"
-        )
+        nonmatchings_dir = "f_nonmat" if args.version == "saturn" else "nonmatchings"
         if not self._overlay:
             nonmatching_index = next(
                 (
                     i
                     for i, part in enumerate(self.relpath.parts)
-                    if part == nonmatchings_dir
+                    if nonmatchings_dir in part
```

Comment:
Why change this?

Reply:
Because:

1. Other places uses it. For example `nonmatchings_dir in c.parts`.
2. If the overlay is `ric` and one of the path parts is `ric_psp`, this will get me the wrong `nonmatching_index` and just produce the wrong output.

Reply:
1. Two different purposes.  Your example simply needs to know if the value is in the list, this is specifically looking for where the value is in the list.
2.  That's literally not possible.  For `part == nonmatchings_dir` to return an index, it has to match exactly.  Meaning it would only return an index if `nonmatchings_dir == "ric_psp"` and `part == "ric_psp"`.  Your change is actually want makes it so this occurs.  If `nonmatchings_dir == "psp"` and `"psp"` is in part, such as `"psp" in "ric_psp"`, then it will give the wrong index.